module0/prework/wsl-computer-setup.md

```diff
@@ -0,0 +1,112 @@
+2. Install WSL
+    * In the Powershell, run `wsl --install`
+    * If asked if you want to allow changes, click 'Yes'
+    * When complete, you should see something like:  
+    <img src="./assets/wsl/WSL-4.png" alt="" style="width: 80%">
+3. Restart you computer
+4. When your computer restarts, Ubuntu should automatically launch
+    <img src="./assets/wsl/WSL-5.png" alt="" style="width: 80%">
+5. When prompted, enter a username (I used my first name) and a password
+    <img src="./assets/wsl/WSL-6.png" alt="" style="width: 80%">
+
+----------------------------------------------------
+
+## Install Ruby
+_All commands will be run in your Ubuntu Terminal_
+1. update the apt package manager with `sudo apt-get update`
+    * If asked for your password, enter the one you configured during the WSL setup above.
+2. install ruby with `sudo apt-get install ruby-full`
+    * enter 'Y' to continue through all prompts
+3. Confirm that ruby has been installed by opening irb (interactive ruby)
+    * `irb`
+    * you can play around with ruby code here!
+    * when you are ready to exit irb, enter `exit!`  
+    <img src="./assets/wsl/WSL-8.png" alt="" style="width: 60%">
+4. Configure your gem settings (don't worry about what gems are right now)
+    * `echo '# Install Ruby Gems to ~/gems' >> ~/.bashrc`
+    * `echo 'export GEM_HOME="$HOME/gems"' >> ~/.bashrc`
+    * `echo 'export PATH="$HOME/gems/bin:$PATH"' >> ~/.bashrc`
+    * `source ~/.bashrc`  
+    <img src="./assets/wsl/WSL-9.png" alt="" style="width: 80%">
+5. Confirm that your gem setup is configured correctly
+    * `gem install pry`
+    * `pry` - this opens an interactive ruby environment similar to irb
+    * `exit!`  
+    <img src="./assets/wsl/WSL-10.png" alt="" style="width: 80%">
+
+---------------------------------------------------------------------
+
+## Install VSCode
+_VSCode is the text editor that we will use for writing code_
+1. Download and Install VSCode
+    * Use the ['Windows Installer'](https://code.visualstudio.com/docs/setup/windows)
+    * Keep all default settings in the install wizard
+2. Close and Restart PowerShell
+    * When you restart, you will need to switch to Ubuntu  
+    <img src="./assets/wsl/WSL-11.png" alt="" style="width: 80%">
+3. Verify setup
+    * create a new file with `touch ruby-playground.rb`
+    * open the file in VSCode with `code ruby-playground.rb`
+    <img src="./assets/wsl/WSL-12.png" alt="" style="width: 80%">
+    * Add the following line to the file and save the file (ctrl+s): `puts "Hello World"`  
+    <img src="./assets/wsl/WSL-13.png" alt="" style="width: 80%">
+    * In PowerShell, run the file with `ruby ruby-playground.rb`  
+    <img src="./assets/wsl/WSL-14.png" alt="" style="width: 80%">
+
+**Congrats! You have created and run your first Ruby program!!**
+
+------------------------------------------------------------------------
+
+## Configure GitHub
```

Comment:
I would probably stop here, and finish with a link to the main prework step to continue w/ GitHub, ssh keys, just so that you don't have to duplicate the content, and it will prevent the instructions for drifting out of sync over time

Reply:
I was hoping that I could do that - my concern was that a few of the commands related to SSH are _slightly_ different from the original setup.  I think it will be more confusing to make a note on the main setup page for these differences.